src/Modules/OrchardCore.Commerce/Abstractions/IProductService.cs

```diff
@@ -14,12 +15,27 @@ public interface IProductService
     /// Returns the products that have the provided SKUs.
     /// </summary>
     Task<IEnumerable<ProductPart>> GetProductsAsync(IEnumerable<string> skus);
+
+    /// <summary>
+    /// Returns the key the product variant is identified by.
+    /// </summary>
+    string GetVariantKey(string sku);
+
+    /// <summary>
+    /// Returns the exact variant of a product, as well as its identifying key, associated with the provided SKU.
+    /// </summary>
+    Task<(PriceVariantsPart Part, string VariantKey)> GetExactVariantAsync(string sku);
 }
 
 public static class ProductServiceExtensions
 {
-    public static async Task<ProductPart> GetProductAsync(this IProductService service, string sku) =>
-        (await service.GetProductsAsync(new[] { sku })).FirstOrDefault();
+    public static int GetDashIndex(string sku) => sku.IndexOf(value: "-", StringComparison.InvariantCulture);
+
+    public static async Task<ProductPart> GetProductAsync(this IProductService service, string sku)
+    {
+        var dashIndex = GetDashIndex(sku);
+        return (await service.GetProductsAsync(new[] { dashIndex == -1 ? sku : sku[..dashIndex] })).FirstOrDefault();
```

Comment:
Use `.SingleOrDefault`, because it should never return more than one results and if it does then that's an exception.